### PR TITLE
Do not append extra pipe if the user adds a closing pipe manually

### PR DESCRIPTION
**Pull request by tk0miya**
### Motivation

If the user adds a closing pipe manually, we currently end up autocompleting in both cases and end up with 3 pipes.

This commits makes it so that if the user inserted a closing pipe to the end of the block argument, we remove the pipe we added automatically and avoid adding yet another one, resulting in two pipes.

refs: #838 

### Implementation

In this scenario, when the user adds a closing pipe manually to the end of the block argument, we end up with two pipes:

1. One in the current position where on type formatting was triggered
2. One in the current position - 1
3. And the opening pipe (and some block arguments)

If we're in this situation, we can remove the pipe we added through autocomplete so that we end up with only one pipe at the end.

### Automated Tests

Added a new example.

### Manual Tests

1. Start the LSP on this branch
2. Type `10.times do` in a file
3. Insert a pipe after the space
4. Verify the pipe is autocompleted
5. Insert a block argument `i`
6. Insert another pipe without moving the cursor
7. Verify that one of the extra pipes is removed and you end up with only one closing pipe

ruby-lsp

### CODE REVIEW

**Comment by tk0miya** [comment]
I have signed the CLA!

File: lib/ruby_lsp/requests/on_type_formatting.rb [hunk 1]
```diff
@@ -69,10 +69,7 @@ def handle_pipe
 
         # If the current character is a pipe and both previous ones are pipes too, then we autocompleted a pipe and the
         # user inserted a third one. In this case, we need to avoid adding a fourth and remove the previous one
-        if line[@position[:character] - 2] == "|" &&
-            line[@position[:character] - 1] == "|" &&
-            line[@position[:character]] == "|"
-
+        if line[..@position[:character]] =~ /do\s+\|[^|]*?\|\|$/
```
Comment (vinistock) [inline lib/ruby_lsp/requests/on_type_formatting.rb:72]: Will this work for one line blocks? E.g.:
```ruby
[].each { || }
```

Also, the comments above this if statement have to be updated given that they no longer reflect the implementation.

File: test/requests/on_type_formatting_test.rb [hunk 2]
```diff
@@ -154,6 +154,39 @@ def test_pipe_is_removed_if_user_adds_manually_after_completion
     assert_equal(expected_edits.to_json, T.must(edits).to_json)
   end
 
+  def test_pipe_is_removed_if_user_adds_manually_after_block_argument
+    document = RubyLsp::Document.new(source: +"", version: 1, uri: "file:///fake.rb")
```
Comment (vinistock) [inline test/requests/on_type_formatting_test.rb:158]: We made a recent change to standardize URI handling, so this needs to become
```suggestion
    document = RubyLsp::Document.new(source: +"", version: 1, uri: URI("file:///fake.rb"))
```

**Comment by vinistock** [comment]
This looks great!

**Review by vinistock — APPROVED** [review]

**Comment by tk0miya** [comment]
Thanks!

**Pull request merged**